utils/firestore.js

```diff
@@ -3539,6 +3539,186 @@ const updateParticipantCorrection = async (participantData) => {
     }
 }
 
+/**
+ * Reset participant survey status
+ * @param {string} connectId - Connect ID of the participant
+ * @param {string} survey - Survey name to reset
+ * @returns {object} - Updated participant document
+ * For now, only ssn survey is supported
+ */
+const resetParticipantSurvey = async (connectId, survey) => { 
+    try {
+        const snapshot = await db.collection('participants').where('Connect_ID', '==', connectId).get();
+        if (snapshot.empty) {
+            throw { message: 'Participant not found.', code: 404 };
+        }
+
+        const participantRef = snapshot.docs[0].ref;
+        const participantData = snapshot.docs[0].data();
+        const { ssnStatusFlag, ssnSurveyStartTime, ssnSurveyCompletedTime, ssnFullGiven, ssnPartialGiven,
+            notStarted, ssnFullGivenTime, ssnPartialGivenTime, no } = fieldMapping;
+
+        // early exit if survey is already in not started status
+        if (participantData[ssnStatusFlag] === notStarted)  {
+            throw {
+                message: 'Failed to reset SSN Survey. The participant\'s SSN survey is "Not Started" status!',
+                code: 400
+            };
+        }
+
+        // Reset participant data
+        if (survey === 'ssn') { // change this to a concept ID
+            await participantRef.update({
+                [ssnStatusFlag]: notStarted,
+                [ssnSurveyStartTime]: FieldValue.delete(),
+                [ssnSurveyCompletedTime]: FieldValue.delete(),
+                [ssnFullGiven]: no,
+                [ssnPartialGiven]: no,
+                [ssnFullGivenTime]: FieldValue.delete(),
+                [ssnPartialGivenTime]: FieldValue.delete(),
+            });
+            const updatedDoc = await participantRef.get();
+            return updatedDoc.data();
+        }
+
+        // Add future surveys here later
+        throw { message: `Survey type ${survey} failed to reset.`, code: 400 };
+
+    } catch (error) {
+        if (error.code) throw error; 
+        console.error('Error resetting participant survey:', error);
+
+        throw {
+            message: `Error resetting participant survey. ${error.message}`,
+            code: 500,
+        };
+        
+    }
+};
+
+/**
+ * Used for NORC Incentive Eligibility tool
+ * Check if participant is eligible for incentive using modified snippet of code from checkDerivedVariables
+ * @param {string} connectId - Connect ID of the participant
+ * @param {string} paymentRound - Payment round to check eligibility for participant
+ * @returns {object} - Object with isEligibleForIncentive boolean and participantData object
+*/
+const checkParticipantForEligibleIncentive = async (connectId, currentPaymentRound) => {
+    try {
```

Comment:
If you wanted to try a simpler approach, you might be able to use the existing function that checks for incentive eligible participants (called from incentive.js) and then see if the returned object contains the participant being checked.